test: refactor tests so they can all be compiled and run in one pass

michaelsbradleyjr opened the pull request:
Make consistent use of the "test helpers" `procSuite` and `asyncTest`.

A couple of test suites are presently disabled: `accounts` and `login_and_logout`. Changes out of scope for this PR need to be introduced and the tests need to be refactored accordingly.

---

Note this PR depends on PR #146.

michaelsbradleyjr committed:
build: bump vendor/nim-waku

michaelsbradleyjr committed:
test: refactor tests so they can all be compiled and run in one pass

Make consistent use of the "test helpers" `procSuite` and `asyncTest`.

A couple of test suites are presently disabled: `accounts` and
`login_and_logout`. Changes out of scope for this PR need to be introduced and
the tests need to be refactored accordingly.

michaelsbradleyjr commented:
@richard-ramos the reason the `login_and_logout` test is disabled in this PR is explained here: https://github.com/status-im/nim-status/blob/test/test_all/test/nim/test_all.nim#L16-L20.

What do you think of moving `web3_conn` into the Status object, like you did with the database connection? I can take a crack at it, just wanted to get your thoughts first.

richard-ramos commented:
Yeah. It makes more sense to have the web3 conn into the status object. We can do that inside the openAccounts PR once it's updated with the changes from this PR

michaelsbradleyjr committed:
feat: introduce a module for nim-status/status-go hybrid procs

michaelsbradleyjr committed:
refactor: nim_status_go Makefile target doesn't depend on migrations

The nim_status_go target builds a pure shim (as a C archive) so it doesn't
depend on migrations, which are only used in the pure/hybrid Nim
implementations.
